assets/js/googlesitekit/api/middleware/preloading.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Custom preloading middleware.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { getStablePath } from '@wordpress/api-fetch/build/middlewares/preloading';
+
+/**
+ * createPreloadingMiddleware
+ *
+ * Based on preloadMiddleware from from @wordpress/api-fetch, this middle is a single-use per-endpoint and provides cached
+ * data for the first request only and any subsequent requests hit the server.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} preloadedData Preloaded data paths.
+ * @param {number} timeout       Timeout value.
+ * @return {Function} Function.
+ */
+function createPreloadingMiddleware( preloadedData, timeout = 1000 ) {
```

Comment:
The added `timeout` parameter is a good idea for testing, but it's not necessary because Jest let's us do all kinds of cool things with its fake timers 😄  For testing, all we want to be able to do is skip the waiting when we say, which Jest allows us to do (see my previous comment). This can be inlined right in the `setTimeout` because the value doesn't need to be variable or used anywhere else.